examples/fiber/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/airbrake/gobrake/v5"
+	fiberbrake "github.com/airbrake/gobrake/v5/fiber"
+	"github.com/gofiber/fiber/v2"
+)
+
+var ProjectId int64 = 999999                               // Insert your Project Id here
+var ProjectKey string = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" // Insert your Project Key here
+
+var notifier = gobrake.NewNotifierWithOptions(&gobrake.NotifierOptions{
+	ProjectId:   ProjectId,
+	ProjectKey:  ProjectKey,
+	Environment: "production",
+})
+
+var availableLocations = []string{
+	"austin",
+	"pune",
+	"santabarbara",
+	"newyork",
+}
+
+// Define models
+
+type Date struct {
+	Date int64 `json:"date,omitempty"`
+}
+
+type WeatherInfo struct {
+	Lat            float32       `json:"lat,omitempty"`
+	Lon            float32       `json:"lon,omitempty"`
+	Timezone       string        `json:"timezone,omitempty"`
+	TimezoneOffset float32       `json:"timezone_offset,omitempty"`
+	Current        interface{}   `json:"current,omitempty"`
+	Minutely       []interface{} `json:"minutely,omitempty"`
+	Hourly         []interface{} `json:"hourly,omitempty"`
+	Daily          []interface{} `json:"daily,omitempty"`
+}
+
+type Error struct {
+	Error string `json:"error,omitempty"`
+}
+
+func main() {
+
+	app := fiber.New()
+	app.Use(fiberbrake.New(notifier))
+	app.Use(TokenAuthMiddleware())
+	// Initialise routes
+	app.Get("/date", getDate)
+	app.Get("/locations", getLocations)
+	app.Get("/weather/:location", getWeather)
+	app.Listen(":3000")
+	defer notifier.Close()
+}
+
+func getDate(c *fiber.Ctx) error {
+	var date Date
+	date.Date = time.Now().Unix()
+	return c.Status(http.StatusOK).JSON(date)
+}
+
+func getLocations(c *fiber.Ctx) error {
+	return c.Status(http.StatusOK).JSON(availableLocations)
+}
+
+func getWeather(c *fiber.Ctx) error {
+	var weatherInfo WeatherInfo
+	location := strings.TrimSpace(c.Params("location")) // It lets you allow whitespaces but case-sensitive
+	if !contains(availableLocations, location) {
+		return c.Status(http.StatusNotFound).JSON(weatherInfo)
+	}
+	weatherInfoFile := location + ".json"
+	weatherInfoFilePath := filepath.Join("assets", weatherInfoFile)
+	// Open our jsonFile
+	jsonFile, err := os.Open(weatherInfoFilePath)
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		notifier.Notify(err, nil) // send error to airbrake
+		return c.Status(http.StatusNoContent).JSON(weatherInfo)
```

Comment:
If we don't have a valid `weatherInfoFilePath`, we shouldn't return `StatusNoContent` (which is a 2xx success message). We should return either `StatusNotFound` or `BadRequest`.